eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2399,6 +2444,19 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     goto(ERR_INFORMATION_LEAK) calling doPublish(localCommitPublished, d.commitments) sending error
   }
 
+  private def handleOutdatedCommitment(channelReestablish: ChannelReestablish, d: HasCommitments) = {
+    nodeParams.outdatedCommitmentStrategy match {
```

Comment:
That pattern match is incomplete, we don't handle the case `HaltIfJustRestarted` case after 10 minutes (so we'll throw a runtime exception).

Reply:
Compiler now catches this, since we have re-enabled the `fatal-warning` flags.

```
match may not be exhaustive.
It would fail on the following input: Stop
    nodeParams.outdatedCommitmentStrategy match {

```